test/test_pipeline_yaml.py

```diff
@@ -792,3 +847,45 @@ def test_save_yaml_overwrite(tmp_path):
     with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
         content = saved_yaml.read()
         assert content != ""
+
+
+def test_save_yaml_with_custom_node_kwargs(tmp_path):
+    class CustomNode(MockNode):
+        def __init__(self, param: int):
+            super().__init__()
+            self.param = param
+
+    pipeline = Pipeline()
+    pipeline.add_node(CustomNode(param=1), name="custom_node", inputs=["Query"])
+    pipeline.save_to_yaml(tmp_path / "saved_pipeline.yml")
```

Comment:
Can't we use `pipeline.get_config()`? Then we wouldn't need to care about loading and saving...

Reply:
I don't see too much value in this as you can achieve the very same with `get_config` and `load_from_config`. Anything else is only testing the YAML serialization.

Reply:
Right. Moving them to `test_pipeline.py` with the other `get_config` tests, I think it makes more sense after this change.